### PR TITLE
Add support for more input types, html/css embeding, mounted method, css scoping

### PR DESCRIPTION
**Pull request by hackal**

### CODE REVIEW

**Comment by hackal** [comment]
@lukasotocerny have a look, what do you think?

File: dist/index.html [hunk 1]
```diff
@@ -104,5 +104,104 @@ <h3>#app3 (methods/if)</h3>
         }
     });
     </script>
+    <hr>
+    <div id="app4">
```
Comment (lukasotocerny) [inline dist/index.html]: add preformatted code of Exp constructor as in examples app1-4

File: dist/index.html [hunk 2]
```diff
@@ -104,5 +104,104 @@ <h3>#app3 (methods/if)</h3>
...
+    var app6 = new Exp({
+        el: '#app7',
+        data: {
+            selected: "25"
+        }
+    });
+    </script>
+
+    <hr>
+    <div id="app8">
```
Comment (lukasotocerny) [inline dist/index.html]: explanation of example, something like <h3>#app8 (style)</h3>

File: dist/index.html [hunk 3]
```diff
@@ -104,5 +104,104 @@ <h3>#app3 (methods/if)</h3>
...
+        <select exp-model="selected">
+            <option value="25">25</option>
+            <option value="35">35</option>
+            <option value="45">45</option>
+        </select>
+        <div exp-bind="selected"></div>
+    </div>
+    <script>
+    var app6 = new Exp({
+        el: '#app7',
```
Comment (lukasotocerny) [inline dist/index.html]: should be consistent numbering

File: src/Exp.js [hunk 4]
```diff
@@ -1,14 +1,26 @@
 class Exp {
     constructor(settings) {
         /* Find application element */
-        this.app = document.querySelector(settings.el);
+        this.el = settings.el || null;
+        this.app = null;
+        this.attatch = settings.attatch || null;
```
Comment (lukasotocerny) [inline src/Exp.js]: attach :)

File: src/Exp.js [hunk 5]
```diff
@@ -1,14 +1,26 @@
...
+        this.html = settings.html || null;
+
+        /* Exp handling styling */
+        this.style = settings.style || null;
+        this.scoped = settings.scoped || false;
 
         this.data = settings.data;
-        this.methods = settings.methods;
+        this.methods = settings.methods || null;
+        this.mounted = settings.mounted || null;
```
Comment (lukasotocerny) [inline src/Exp.js]: is mounted something we need?

Comment (lukasotocerny) [reply src/Exp.js]: yes

**Pull request merged**